apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
Okay, will do that.

> That's why I think it makes sense to have the 2 events (and possibly promises). Basically some parts of the code will need to wait only 1, but some other parts will need to wait until 2.

Yeah, with DraftService we'll have promises anyway. About ```stored``` event - yep, there are potential use cases for this, though we don't have any yet :) I think we can add ```stored``` event only when we do ```DraftsService``` and decide that we need such event :)